### ISSUE TITLE
macros do not use inheritance

### ISSUE THREAD

**Issue by mousewax** [ArnoldSmith86/virtualtabletop#1561]
I attempted to set properties on widgets using the macro editor.
```if(w.component== 'worker') {
  w.resetProperties={
    "x":w.x,
    "y":w.y
  }
}```

One widget has:
```{
  "id": "player1worker1",
  ...
  "component": "worker"
}```

The others inherit from this main worker:
```{
  "id": "player1worker2",
  ...
  "inheritFrom": "player1worker1"
}```

Only `player1worker1` was updated with the new reset properties.

**Comment by robartsd** [comment]
Yes, I believe macros work with just the JSON state object, not the fully implemented widget, so they don't look at default values (inherited from other widgets or otherwise).

**Comment by ArnoldSmith86** [comment]
> Yes, I believe macros work with just the JSON state object, not the fully implemented widget, so they don't look at default values (inherited from other widgets or otherwise).

This.

You could do `if(w.id. match(/worker/))` or `if(w.inheritFrom == 'xyz')`..

**Comment by robartsd** [comment]
I'm leaning towards closing this issue - behaves as intended. Fully implemented widgets can be modified with routine code using SELECT and SET.

**Comment by 96LawDawg** [comment]
I agree. I did try to help mousewax with a routine but Raphael was able to create a simple macro faster than I could fix my routine. But I don’t think this issue needs fixing.

**Comment by RaphaelAlvez** [comment]
It's not a matter of working as intended. This is mostly a request to change the behavior.

At least it was for me when I asked for mouse wax to open the issue
